src/main/java/ch/njol/skript/expressions/ExprProtocolVersion.java

```diff
@@ -0,0 +1,120 @@
+		"For more information and list of protocol versions <a href='http://wiki.vg/Protocol_version_numbers'>visit wiki.vg</a>.",
+		"If this protocol version doesn't match with the protocol version of the client, the client will see the <a href='#ExprVersionString'>version string</a>.",
+		"But please note that, this expression has no visual effect over the version string. " +
+		"For example if the server uses PaperSpigot 1.12.2, and you make the protocol version 107 (1.9), the version string will not be \"Paper 1.9\", it will still be \"Paper 1.12.2\". " +
+		"But then you can customize the <a href='#ExprVersionString'>version string</a> as you wish.",
+		"Also if protocol version of the player is higher than protocol version of the server, it will say \"Server out of date!\", and if vice-versa \"Client out of date!\" when you hover on the ping bars.",
+		"",
+		"This can be set in a <a href='events.html#server_list_ping'>server list ping</a> event only (increase and decrease effects cannot be used because doesn't make sense).",
+		"",
+		"Requires PaperSpigot 1.12.2+."})
+@Examples({"on server list ping:",
+		"	set the protocol version to 0 # 13w41a (1.7), so it will show the version string always"})
+@Since("INSERT VERSION")
+public class ExprProtocolVersion extends SimpleExpression<Number> {
+
+	static {
+		Skript.registerExpression(ExprProtocolVersion.class, Number.class, ExpressionType.SIMPLE, "[the] [(sent|required|fake)] protocol version [number]");
+	}
+
+	@SuppressWarnings("null")
+	private Kleenean delay;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		boolean isServerPingEvent = ScriptLoader.isCurrentEvent(ServerListPingEvent.class);
+		boolean isPaperEvent = Skript.classExists("com.destroystokyo.paper.event.server.PaperServerListPingEvent") && ScriptLoader.isCurrentEvent(PaperServerListPingEvent.class);
+		if (isServerPingEvent) {
+			Skript.error("The protocol version expression requires PaperSpigot 1.12.2+");
+			return false;
+		} else if (!isPaperEvent) {
+			Skript.error("The protocol version expression can't be used outside of a server list ping event" + (isServerPingEvent ? " and requires PaperSpigot 1.12.2+" : ""));
+			return false;
+		}
+		delay = isDelayed;
+		return true;
+	}
+
+	@Override
+	public boolean isSingle() {
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public Number[] get(Event e) {
+		return CollectionUtils.array(((PaperServerListPingEvent) e).getProtocolVersion());
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (delay == Kleenean.TRUE) {
+			Skript.error("Can't change the protocol version anymore after the server list ping event has already passed");
+			return null;
+		}
+		if (mode == Changer.ChangeMode.SET)
+			return CollectionUtils.array(Number.class);
```

Comment:
Would it be possible and would it have any real use case to somehow allow resetting the protocol version here? I mean, in case some other plugin changes it.. but on the other way most plugins that change the protocol version probably just do that directly with packets, so I don't really know. Opinion?